internal/cmd/oohelperd/internal/nwcth/http.go

```diff
@@ -0,0 +1,110 @@
+package nwcth
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"net/http/cookiejar"
+	"net/url"
+	"strings"
+
+	"github.com/ooni/probe-cli/v3/internal/iox"
+	"github.com/ooni/probe-cli/v3/internal/runtimex"
+)
+
+// HTTPConfig configures the HTTP check.
+type HTTPConfig struct {
+	Jar       *cookiejar.Jar
+	Headers   map[string][]string
+	Transport http.RoundTripper
+	URL       *url.URL
```

Comment:
You should document which fields are mandatory and which are optional. For example, the `Jar` seems to be optional and also `Headers` seem optional, while `Transport` seems mandatory. If something is optional, document what happens if it's not provided. I'd also like to see the generic type for `Jar` (i.e., the interface in `net/http`) being used as opposed to the concrete type.

Hint that may be useful (later) to implement testing: if, say, `Transport` is optional and you make it optional and you construct a suitable `Transport` if it's not set, you probably have minimized the scope of `Transport`, reduced what we called bureaucracy, and reduced the amount of code you need to write in the caller, isn't it?

Reply:
(It may be tempting to use `RequestHeaders` and `RequestURL` for additional clarity. Sometimes I'm a bit more brief with naming, especially when there is inline docs such as

```Go
// Headers contains the request headers
Headers map[string][]string
```

because one can hover with the mouse and see what a field is. For this reason, it may be smart to provide a small piece of documentation to any field of any structure. Hovering can be very useful when navigating code.)

Reply:
I'm not sure about making Transport optional. I feel like it is more explicit to set the Transport in the call to HTTPDo, as it identifies whether it is a TCP or QUIC based transport. Otherwise, I guess an HTTPS (so TCP) Transport should be the default?

Reply:
Good point! This is not a case where having a default would make things simpler. Let's leave it as is, then!